cpp/src/arrow/compute/util.cc

```diff
@@ -35,7 +35,7 @@ void TempVectorStack::alloc(uint32_t num_bytes, uint8_t** data, int* id) {
   int64_t new_top = top_ + PaddedAllocationSize(num_bytes) + 2 * sizeof(uint64_t);
   // Stack overflow check (see GH-39582).
```

Comment:
Could you move this comment to `CheckAllocSizeValid()`?

Reply:
done